src/main/java/io/kokuwa/keycloak/metrics/store/MicrometerEventStoreProvider.java

```diff
@@ -0,0 +1,121 @@
+        super.onEvent(event, includeRepresentation);
+        var isEventsMetricsEnabled = CommunityProfiles.isEventsMetricsEnabled();
+        if (isEventsMetricsEnabled) {
+            addAdminEvent(event);
+        }
+    }
+
+    @Override
+    public void close() {
+    }
+
+    private String getRealmName(String id) {
+        return Optional.ofNullable(session.getContext()).map(KeycloakContext::getRealm)
+                .filter(realm -> id == null || id.equals(realm.getId()))
+                .or(() -> {
+                    log.tracev("Context realm was empty with id {0}", id);
+                    return Optional.ofNullable(id).map(session.realms()::getRealm);
+                })
+                .map(RealmModel::getName)
+                .orElseGet(() -> {
+                    log.warnv("Failed to find realm with id {0}", id);
+                    return id;
+                });
+    }
+
+    private String toBlank(Object value) {
+        return value == null ? "" : value.toString();
+    }
+
+    void addEvent(Event event) {
+        session
+                .getTransactionManager()
+                .enlistAfterCompletion(
+                        new AbstractKeycloakTransaction() {
+                            @Override
+                            protected void commitImpl() {
+                                KeycloakModelUtils.runJobInTransaction(
+                                        session.getKeycloakSessionFactory(),
+                                        (s) -> {
+                                            Metrics.counter("keycloak_event_user",
+                                                            "realm", toBlank(replaceIds ? getRealmName(event.getRealmId()) : event.getRealmId()),
+                                                            "type", toBlank(event.getType()),
+                                                            "client", toBlank(event.getClientId()),
+                                                            "error", toBlank(event.getError()))
+                                                    .increment();
+                                        });
+                            }
+
+                            @Override
+                            protected void rollbackImpl() {
+                            }
+                        });
+    }
+
+    void addAdminEvent(AdminEvent event) {
+        session
+                .getTransactionManager()
+                .enlistAfterCompletion(
+                        new AbstractKeycloakTransaction() {
+                            @Override
+                            protected void commitImpl() {
+                                KeycloakModelUtils.runJobInTransaction(
+                                        session.getKeycloakSessionFactory(),
+                                        (s) -> {
+                                            Metrics.counter("keycloak_event_admin",
```

Comment:
Duplicates what happens in `MetricsEventListener`. Todo abstract out or use the listener directly.